File: client/core/src/main/kotlin/me/grian/player/Player.kt
package me.grian.player

import com.badlogic.gdx.Gdx
import me.grian.Main
import me.grian.network.NetworkManager
import me.grian.network.packets.c2s.C2SMovePacket

class Player(
    val pos: Point,
    var chunkPos: Point,
    var realX: Float,
    var realY: Float,
    var name: String
) {
    fun move(x: Int, y: Int) {
        // TODO: bounds check on map once i have that sorted out
        if (x !in 0..15 || y !in 0..31) return
        if (Main.players.any { x == it.pos.x && y == it.pos.y }) return

        // maybe reconsider sending packet on function call, not sure though, have to see which one
        // ends up being more used, setting x/y directly or normnal move
        NetworkManager.sendPacket(
            C2SMovePacket(x, y)
        )
    }

    constructor(x: Int, y: Int, chunkX: Int, chunkY: Int,  name: String):
        this(Point(x, y), Point(chunkX, chunkY), x * Main.tileSize, y * Main.tileSize, name)
}

File: client-go/game/player.go
package game

type Player struct {
	X     int32
	Y     int32
	RealX int32
	RealY int32
	Name  string
}

func (p *Player) Move(newX, newY int32, game *Game) {
	// TODO: send packet
	// TODO: bounds after map loading
	// basic bounds checking for testing purposes
	if newX > 15 || newX < 0 || newY > 15 || newY < 0 {
		return
	}

	p.X = newX
	p.Y = newY

	p.RealX = p.X * game.TileSize
	p.RealY = p.Y * game.TileSize
}
